source/iterator/cdc.go

```diff
@@ -96,6 +97,17 @@ func (c *CDCIterator) HasNext(ctx context.Context) (bool, error) {
 	c.currentBatch, err = c.snowflake.GetTrackingData(ctx, getStreamName(c.table),
 		getTrackingTable(c.table), c.columns, c.offset, c.batchSize)
 	if err != nil {
+		// Snowflake library sends request to abort query with query and to server when get context cancel.
+		// But sometimes query was executed or didn't start execution.
+		// On this case snowflake server return specific error:
+		// 000605: Identified SQL statement is not currently executing.
+		// Connector can't return this error and connector replace to
+		// context cancel error
+		// https://github.com/snowflakedb/gosnowflake/blob/master/restful.go#L449
```

Comment:
Let me check if I understood this correctly: If the context is canceled, you get this error `000605: Identified SQL statement is not currently executing`? Why does it matter if "sometimes query was executed or didn't start execution"?

Reply:
I want to add information why  we get this error, but if you think it is redundant , we can remove it

Reply:
Yes we will know, snowflake library return this error only inside cancelQuery function
https://github.com/snowflakedb/gosnowflake/blob/master/restful.go#L412
https://github.com/snowflakedb/gosnowflake/blob/master/restful.go#L30

CancelQuery executes only if we have context cancel or context deadline errors
https://github.com/snowflakedb/gosnowflake/blob/master/restful.go#L182

Reply:
Thanks for that! In that case, it might make sense to extract this code into a single function, which we can call from multiple places (because it's always the same behavior).